Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/ChangeAffectedInputTests.cs

```diff
@@ -74,6 +74,93 @@ public void DirectAffectedFileInputTest()
+            FileArtifact cOutput = CreateOutputFileArtifact(root: dirPath, prefix: "pip-c-out-file");
+            AbsolutePath changeAffectedWrittenFile = CreateUniqueObjPath("change");
+            var pipBuilderC = CreatePipBuilder(new[]
+            { 
+                // Ensure that pip reads the changeAffectedWrittenFile.
+                Operation.ReadFile(FileArtifact.CreateSourceFile(changeAffectedWrittenFile), doNotInfer: true),
+                Operation.ReadFile(aOutput),
+                Operation.ReadFile(bOutput),
+                Operation.WriteFile(cOutput)
+            });
+            pipBuilderC.SetChangeAffectedInputListWrittenFile(changeAffectedWrittenFile);
+            var pipC = SchedulePipBuilder(pipBuilderC);
+            Configuration.Schedule.InputChanges = changeList.Path;
+
+            // Build1 with change in aInput
+            File.WriteAllText(changeList.Path.ToString(Context.PathTable), aInput.Path.ToString(Context.PathTable));
+            var result = RunScheduler();
+            result.AssertCacheMiss(pipA.Process.PipId, pipB.Process.PipId, pipC.Process.PipId);
+            var actualAffectedInput = File.ReadAllText(changeAffectedWrittenFile.ToString(Context.PathTable));
+            var expectedAffectedInput = aOutput.Path.GetName(Context.PathTable).ToString(Context.PathTable.StringTable);
+            XAssert.AreEqual(expectedAffectedInput, actualAffectedInput);
+
+            ResetPipGraphBuilder();
+
+            // Build2 with change in bInput
+            File.WriteAllText(bInput.Path.ToString(Context.PathTable), "pipBChange");
+            File.WriteAllText(changeList.Path.ToString(Context.PathTable), bInput.Path.ToString(Context.PathTable));
+            pipA = SchedulePipBuilder(pipBuilderA);
+            pipB = SchedulePipBuilder(pipBuilderB);
+            pipC = SchedulePipBuilder(pipBuilderC);
+
+            result = RunScheduler();
+            result.AssertCacheHit(pipA.Process.PipId);
+            result.AssertCacheMiss(pipB.Process.PipId, pipC.Process.PipId);
+
+            actualAffectedInput = File.ReadAllText(changeAffectedWrittenFile.ToString(Context.PathTable));
+            expectedAffectedInput = bOutput.Path.GetName(Context.PathTable).ToString(Context.PathTable.StringTable);
+            XAssert.AreEqual(expectedAffectedInput, actualAffectedInput);
+
+            ResetPipGraphBuilder();
+
+            // Build3 revert change in bInput
+            File.WriteAllText(bInput.Path.ToString(Context.PathTable), "pipBOrigin");
+            File.WriteAllText(changeList.Path.ToString(Context.PathTable), bInput.Path.ToString(Context.PathTable));
+            pipA = SchedulePipBuilder(pipBuilderA);
+            pipB = SchedulePipBuilder(pipBuilderB);
+            pipC = SchedulePipBuilder(pipBuilderC);
+
+            result = RunScheduler();
+            result.AssertCacheHit(pipA.Process.PipId, pipB.Process.PipId);
+            result.AssertCacheMiss(pipC.Process.PipId);

```

Comment:
>result.AssertCacheMiss(pipC.Process.PipId); [](start = 12, length = 43)

Give comment why you expect to have cache miss here.

Reply:
will do